### ISSUE TITLE
Implement Intake catalogs for search results

### ISSUE THREAD

**Issue by matthewhanson** [pangeo-data/intake-stac#11]
Currently the plugin allows one to open existing static catalog and crawl it to open up STAC Items as an Intake catalog of assets.

In many cases though, users will want to take the output of a search result from a STAC API and open up all those Items as an Intake catalog of catalogs of assets.

The STAC `Items` class can be used here, which is what the [sat-search](https://github.com/sat-utils/sat-search) library uses.   Not sure if the sat-search dependency provides additional features that would make it worthwhile as a dependency as well.

Additionally, the `Items` class provides a way to save and load a STAC catalog as a single file that is a FeatureCollection with additional top level fields containing the collections. This approach will probably make it into STAC next version: https://github.com/radiantearth/stac-spec/issues/385 so this becomes just adding the ability to open up either a traditional STAC catalog or a single file catalog.

**Comment by matthewhanson** [comment]
With the current dev version of STAC, [ItemCollections](https://github.com/radiantearth/stac-spec/blob/dev/item-spec/itemcollection-spec.md) have been added, which are basically just FeatureCollections of Items.

The new [single-file-stac extension](https://github.com/radiantearth/stac-spec/tree/dev/extensions/single-file-stac) extends the ItemCollection to include Collections so that it is a self-contained catalog. 

So intake STAC should be able to read a catalog, in either the original linked file format, or the single file format (which is also what sat-search saves results as).

**Comment by jhamman** [comment]
I worked on this a bit today (see #29). The basic idea is as follows:

##### use sat-search to get a `satstac.ItemCollection`:
```python
results = satsearch.Search.search(
    collection='landsat-8-l1',
    bbox=bbox,
    sort=['<datetime'], #earliest scene first
    property=properties)
items = results.items()
```

##### use intake-stac to load the feature collection as a catalog:
```python
cat = intake.open_stac_item_collection(items)
```

`cat` ends up being an `intake.Catalog` where each item is mapped to an entry (just like the rest of the Intake-Stac catalog objects).

**Comment by jhamman** [comment]
closed by #29
